File: Fachreferat/Fachreferat/TwitterUser.cs
using System;
using System.Collections.Generic;

namespace DataMining
{
    public class TwitterUser
    {
        public string Id { get; set; }
        public string Name { get; set; }
        public string Username { get; set; }
        public PublicMetricsUser Public_metrics { get; set; }
        public KMeansProperties KMeansProps { get; set; }
    }

    public class PublicMetricsUser
    {
        public int Followers_count { get; set; }
        public int Following_count { get; set; }
        public int Tweet_count { get; set; }
        public int Listed_count { get; set; }
    }

    public class KMeansProperties
    {
        public KMeansProperties(double x, double y, int centroidId)
        {
            X = x;
            Y = y;
            CentroidId = centroidId;
        }

        public double X { get; set; }
        public double Y { get; set; }
        public int CentroidId { get; set; }
    }

    public class TwitterUserDto
    {
        public List<TwitterUser> data { get; set; }
    }
}

File: Fachreferat/Fachreferat/ClientApp/src/model/TwitterUser.ts
export interface ITwitterUser{
    id: string;
    name: string;
    username: string;
    public_metrics: IPublicMetricsUser;
    kMeansProps: IKMeansProps;
}

export interface IPublicMetricsUser{

    followers_count: number;
    following_count: number;
    tweet_count : number;
    listed_count : number;

}

export interface IKMeansProps{
    x: number;
    y: number;
    centroidId: number;
}


export interface IClusterGraphProps{
    name: string
    x: number;
    y: number;
    centroidId: number;
    shape: number
}


export function creatClusterGraphProps(name: string, x: number, y: number, centroidId: number, isCenterOfCentroid :boolean = false ):IClusterGraphProps {
    return Object.assign(
        {
            name: name,
            x: x,
            y: y,
            centroidId: centroidId,
            shape: isCenterOfCentroid ? 1: 0
            
        }
    );
}